app/models/foreman_docker/service/actions/container/pull.rb

```diff
@@ -0,0 +1,44 @@
+module ForemanDocker
+  module Service
+    module Actions
+      module Container
+        class Pull < ::Actions::EntryAction
+          def plan(container)
+            action_subject(container)
+            plan_self(:container => container.id)
+          end
+
+          def run(event = nil)
+            case event
+            when :done # do nothing and move to next step
+            else
+              suspend do |suspended_action|
+                Thread.new do
+                  begin
+                    container = ::Container.find(input[:container])
+                    container.compute_resource
+                      .create_image(:fromImage => container.repository_pull_url)
+                    suspended_action << :done
+                  rescue => e
+                    action_logger.error(e.message)
+                    action_logger.debug(e.backtrace.join("\n "))
+                    error!(e)
```

Comment:
This is not a thread save call, changing the state of the action from a different thread. The error needs to be sent back to the action same as `:done` event.